presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -323,6 +332,10 @@ public ConnectorPageSource createPageSource(
                     delegate = new HiveCoercionRecordCursor(regularAndInterimColumnMappings, typeManager, delegate);
                 }
 
+                if (bucketAdaptation.isEmpty() && bucketValidator.isPresent()) {
+                    delegate = bucketValidator.get().toRecordCursor(delegate, typeManager);
```

Comment:
Maybe add a comment why this is only when `bucketAdaptation.isEmpty()`
also, even in case of `bucketAdaptation.isEmpty()` we should be able to do some validation. How hard would it be? A followup issue?

Reply:
It's not needed because the adapter already does the same validation.

Reply:
please add this as a code comment maybe

Reply:
Done